### ISSUE TITLE
forceboard data

### ISSUE THREAD

**Issue by adrianlzt** [Stevie-Ray/hangtime-grip-connect#2]
Hi!

Just found your repo looking for a way to export data from the forceboard device.
I see in the readme it is marked as work in progress and there is already some code to get battery and humidity info.

Do you have any estimation to implement the forceboard?

I will happily help but I do not have the device (a friend has it, but I do not have physical access).

I have decompiled the apk and its not ofuscated, so it's quite useful (3a90328c-c266-4c76-b05a-6af6104a0b13 looks like is temperature).

Thanks!

**Comment by Stevie-Ray** [comment]
Hi @adrianlzt , 

Thanks for you interest in the project!

I've connected Matt Cupal the founder of PitchSix and he told me that he's happy to provide an API to integrate in. They haven’t published it yet, but will "soon".

I do own the device, and was playing around with it using [WireShark](https://www.wireshark.org/ ) and found out how to read data from the force sensor but haven't been able to decode it. I'll look into it again when I have some time. 

Same for temperature, if you have some info on how to read the data please let me know! Also do you know what 467a8516-6e39-11eb-9439-0242ac130002 does?

**Comment by adrianlzt** [comment]
Nice :)

My approach to reverse engineer this things usually pass to decompile the [apk]](https://apkpure.com/force-board/com.pitchsix.forceboard/download) with [apktool](https://apktool.org/) and read the `.smali` files.

In this case I think the interesting file is [BluetoothLeService.smali.txt](https://github.com/user-attachments/files/17292266/BluetoothLeService.smali.txt).

Smali is a pseudo-code that looks like java.

For that UUID I find this things:

```
.method public writeBeginSimulationModeCharacteristic()V
    .locals 3

    .line 716
    const-string v0, "467a8516-6e39-11eb-9439-0242ac130002"
```

```
.method public writeEnterLiveScreenCharacteristic(Z)V
    .locals 3
    .annotation system Ldalvik/annotation/MethodParameters;
        accessFlags = {
            0x0
        }
        names = {
            "isEnterInLive"
        }
    .end annotation

    .line 952
    const-string v0, "467a8516-6e39-11eb-9439-0242ac130002"
```

```
.method public writeLiveBeginRepCharacteristic()V
    .locals 4

    .line 743
    const-string v0, "467a8516-6e39-11eb-9439-0242ac130002"
```

I hope this could be useful.

P.S.: just out of curiosity, how are you capturing the bluetooth traffic? In the android phone doing the `Enable Bluetooth HCI snoop log`?

**Comment by Stevie-Ray** [comment]
Interesting! I'll follow your method and see what I can find 😄 

Yes I enable `Bluetooth HCI snoop logging` and use WireShark to live debug  the device over USB and see what payload is send to a characteristics UUID in de app. 

If you have access to the device in the meantime, a PR is more than welcome!

**Comment by Stevie-Ray** [comment]
Hi @adrianlzt in v0.5.4 I've added support for Stream, Stop & Temperature

Give it a try! https://stevie-ray.github.io/hangtime-grip-connect/

**Comment by adrianlzt** [comment]
The version deployed on vercel, https://grip-connect.vercel.app/, has the
latest version?

**Comment by Stevie-Ray** [comment]
Yeah, it deploys a new version on every commit. You can also use it in [HangTime](https://play.google.com/store/apps/details?id=nl.stevie.ray.hangtime) on Android. 

![screenshot](https://github.com/user-attachments/assets/f9174cd8-ef69-4ebc-9103-6e9a828a4d26)

**Comment by adrianlzt** [comment]
Looks like the download button is not working. I have tested with Progressor and it works perfectly.

Sorry no to provide more info, I'm just trying to be an useful intermediator :)

**Comment by Stevie-Ray** [comment]
@adrianlzt  No problem at all, thanks for letting me know. The issue with the download button should be resolved in the latest build (v0.5.5). There's still some work needed for the download functionality when connecting multiple devices, but it should now work fine with a single device.

**Comment by Stevie-Ray** [comment]
@adrianlzt Docs on the API: [Force Board Portable Public API](https://cdn.shopify.com/s/files/1/0249/5525/6922/files/Force_Board_Public_API_1.0.pdf)
